src/dmd/expressionsem.d

```diff
@@ -5225,12 +5225,12 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             else if (sc.func)
             {
                 // https://issues.dlang.org/show_bug.cgi?id=11720
-                // include Dataseg variables
                 if ((s.isFuncDeclaration() ||
                      s.isAggregateDeclaration() ||
                      s.isEnumDeclaration() ||
                      s.isTemplateDeclaration() ||
-                     v && v.isDataseg()) && !sc.func.localsymtab.insert(s))
+                     v
```

Comment:
Maybe put this first in the order of conditions, to keep it in order of least to most expensive.